x/mongo/driver/topology/topology_test.go

```diff
@@ -230,6 +232,29 @@ func TestServerSelection(t *testing.T) {
 			t.Fatalf("did not receive error from server selection")
 		}
 	})
+	t.Run("Context Deadline Error", func(t *testing.T) {
+		desc := description.Topology{}
+		topo, err := New()
+		noerr(t, err)
+		subCh := make(chan description.Topology, 1)
+		subCh <- desc
+		resp := make(chan error)
+		timeout := make(chan time.Time)
+		callback := func() {
+			state := newServerSelectionState(selectContextError, timeout)
```

Comment:
Using the `selectContextError` selector is fine here, but it's more realistic for this error to happen in the actual server selection loop rather than the selector. You can use the the `Cancel` test above for inspiration. It uses the `selectNone` selector, so server selection will loop and then timeout after 50ms because that's your context timeout. If you do this, you can also remove the `selectContextError` selector if it's unused.